### ISSUE TITLE
OpenCV 4.x+ requires enabled C++11 support ROS2 Foxy

### ISSUE THREAD

**Issue by blakermchale** [leggedrobotics/darknet_ros#361]
**Describe the bug**
When I build the foxy branch of `darknet_ros` with CUDA support I get the following errors:
```
/usr/include/opencv4/opencv2/core/cvdef.h:704:4: error: #error "OpenCV 4.x+ requires enabled C++11 support"
  704 | #  error "OpenCV 4.x+ requires enabled C++11 support"
      |    ^~~~~
/usr/include/opencv4/opencv2/core/cvdef.h:710:10: fatal error: array: No such file or directory
  710 | #include <array>
```

**To Reproduce**
1. Clone the foxy branch
2. `colcon build -DCMAKE_BUILD_TYPE=Release`
3. See error

**Expected behavior**
I expect ROS2 Foxy to compile with CUDA support.

**System (please complete the following information):**
 - OS: Ubuntu 20.04
 - ROS version: Foxy
 - GPU when relevant: NVIDIA RTX 2080 SUPER

**Additional context**
I noticed that this problem was fixed for Noetic in #308.

**Comment by blakermchale** [comment]
Just saw #257 and the discussion at the end that referenced this fork https://github.com/ajaypaul2008/darknet_ros/tree/foxy

**Comment by Ar-Ray-code** [comment]
Please try to use [my fork](https://github.com/Ar-Ray-code/darknet_ros_fp16) :) .

**Comment by blakermchale** [comment]
Thanks! I switched to your branch on the PR you made and it works well!

**Comment by Ar-Ray-code** [comment]
Please close this.

**Comment by blakermchale** [comment]
Closed since #319 will fix this once it is merged.
